src/platforms/hosted/serial_unix.c

```diff
@@ -43,6 +47,73 @@ static uint8_t read_buffer[READ_BUFFER_LENGTH];
 static size_t read_buffer_fullness = 0U;
 static size_t read_buffer_offset = 0U;
 
+/* Socket code taken from https://beej.us/guide/bgnet/ */
+static bool try_opening_network_device(const char *const name)
+{
+	// Maximum legal length of a hostname
+	char hostname[256];
+
+	// The service name or port number
+	char *service_name;
+
+	struct addrinfo addr_hints = {
+		.ai_family = AF_UNSPEC,
+		.ai_socktype = SOCK_STREAM,
+		.ai_protocol = IPPROTO_TCP,
+		.ai_flags = AI_ADDRCONFIG | AI_V4MAPPED,
+	};
+	struct addrinfo *server_info;
+
+	if (!name) {
+		return false;
+	}
```

Comment:
The braces on this are superfluous and can be dropped. This check should ideally be done at the top of the function - we are in C11, so there is no requirement that variables are defined first (it actually benefits stack usage and such for them to be declared as needed). Please include a length check on `name` to warn users of truncation given the strncpy() into a stack buffer.

Reply:
My Rust is showing here.